spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/mustache/web/MustacheViewResolver.java

```diff
@@ -51,20 +53,30 @@ public void setCompiler(Compiler compiler) {
 		this.compiler = compiler;
 	}
 
+        /**
+         *
+         * @param charset the charset to set. Used for reading resources files
+         */
+        public void setCharset(String charset) {
+            this.charset = charset;
+        }
+
 	@Override
 	protected View loadView(String viewName, Locale locale) throws Exception {
 		Resource resource = resolveResource(viewName, locale);
 		if (resource == null) {
 			return null;
 		}
-		MustacheView view = new MustacheView(createTemplate(resource));
+
+		MustacheView view = (MustacheView) buildView(viewName);
+                view.setTemplate(createTemplate(resource));
 		view.setApplicationContext(getApplicationContext());
 		view.setServletContext(getServletContext());
 		return view;
 	}
 
 	private Template createTemplate(Resource resource) throws IOException {
-		return this.compiler.compile(new InputStreamReader(resource.getInputStream()));
+		return this.compiler.compile(new InputStreamReader(resource.getInputStream(), Charset.forName(charset)));
```

Comment:
The compiler uses `MustacheResourceTemplateLoader` internally and that one is configured to use the configured charset.